### ISSUE TITLE
Prebuilt mozjpeg binary doesn't work with 7.0.0 on Ubuntu

### ISSUE THREAD

**Issue by thasmo** [imagemin/imagemin-mozjpeg#26]
Installing `imagemin-mozjpeg@6.0.0` logs `mozjpeg pre-build test passed successfully`, using `imagemin-mozjpeg@7.0.0` logs:
```
  ⚠ The `/.../node_modules/mozjpeg/vendor/cjpeg` binary doesn't seem to work correctly
  ⚠ mozjpeg pre-build test failed
  ℹ compiling from source
  ✖ Error: autoreconf -fiv && ./configure --disable-shared --disable-dependency-tracking --with-jpeg8  --prefix="/.../node_modules/mozjpeg/vendor" --bindir="/.../node_modules/mozjpeg/vendor" --libdir="/.../node_modules/mozjpeg/vendor" && make -j8 && make install -j8
Command failed: ./configure --disable-shared --disable-dependency-tracking --with-jpeg8  --prefix="/.../node_modules/mozjpeg/vendor" --bindir="/.../node_modules/mozjpeg/vendor" --libdir="/.../node_modules/mozjpeg/vendor"
configure: error: no nasm (Netwide Assembler) found
```

**Comment by thasmo** [comment]
Oh, I guess this related to https://github.com/imagemin/mozjpeg-bin/issues/33 - feel free to close if applicable.

**Comment by Daxy24** [comment]
The `/.../source/node_modules/mozjpeg/vendor/cjpeg` binary doesn't seem to work correctly
 
 
  ⚠ mozjpeg pre-build test failed
 
 
  ℹ compiling from source
 
  ✖ Error: autoreconf -fiv && ./configure --disable-shared --disable-dependency-tracking --with-jpeg8  --prefix="/.../source/node_modules/mozjpeg/vendor" --bindir="/.../source/node_modules/mozjpeg/vendor" --libdir="/.../source/node_modules/mozjpeg/vendor" && make -j1 && make install -j1
Command failed: autoreconf -fiv
/bin/sh: 1: autoreconf: not found
 
    at ChildProcess.exithandler (child_process.js:206:12)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:877:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)

**Comment by windware** [comment]
It needs,
autoconf libtool pkg-config nasm
to compile but somehow it builds a shared binary requiring libpng16-dev as well.

**Comment by erick2014** [comment]
For any reason My ubuntu was missing some packages to compile, I firs had to add a new gpg key:

`wget -qO - https://raw.githubusercontent.com/yarnpkg/releases/gh-pages/debian/pubkey.gpg | sudo apt-key add -`

then sudo apt-get install  autoconf libtool pkg-config nasm build-essential

and now it compiles the module

**Comment by DevSide** [comment]
sudo apt-get install libpng16-dev

**Comment by barryvanveen** [comment]
Is the dependency upon `libpng16-dev` intentional or incidental? 

On our build servers we cannot install `libpng16-dev` because `libpng12-dev` is already present. This is currently holding us back from upgrading packages that rely on imagemin-mozjpeg. 

Does anyone have a suggestion on how to solve this? I might, for example, be able to compile `libpng16-dev` from source, but how can I instruct `imagemin-mozjpeg` to use that compiled version? 

Any help would be greatly appreciated!

**Comment by feryardiant** [comment]
I'm also having same issue and can be fixed just by `apt install nasm` since `autoconf libtool pkg-config nasm build-essential` are already installed on my Ubuntu 16.04 (WSL).

```
> mozjpeg@5.0.0 postinstall /path/to/project/node_modules/mozjpeg
> node lib/install.js

  ⚠ The `/path/to/project/node_modules/mozjpeg/vendor/cjpeg` binary doesn't seem to work correctly
  ⚠ mozjpeg pre-build test failed
  ℹ compiling from source
  ✔ mozjpeg built successfully
```

**Comment by JimmyMultani** [comment]
Updating to `mozjpeg@6.0.0` allowed us to successfully build on our Ubuntu server. Will this be part of a version update soon?

**Comment by dan-rally** [comment]
Can confirm ~6.0.0 works for me too, ~7.0.0 binary doesn't work and I am not able to install the various build tools etc. Thanks @JimmyMultani

**Comment by ftreguer** [comment]
I think it's fixed in 8.0.0. I've tested it today, it works for me. Build fails with 6.0.0 and 7.0.0 right now but works fine in 8.0.0
